pandas/core/series.py

```diff
@@ -3800,14 +3806,31 @@ def to_csv(self, path=None, index=True, sep=",", na_rep='',
             Character recognized as decimal separator. E.g. use ',' for
             European data
         """
+
         from pandas.core.frame import DataFrame
         df = DataFrame(self)
-        # result is only a string if no path provided, otherwise None
-        result = df.to_csv(path, index=index, sep=sep, na_rep=na_rep,
-                           float_format=float_format, header=header,
-                           index_label=index_label, mode=mode,
-                           encoding=encoding, compression=compression,
-                           date_format=date_format, decimal=decimal)
+
+        if path is not None:
+            warnings.warn("The signature of `Series.to_csv` will be "
+                          "aligned to that of `DataFrame.to_csv` in the "
+                          "future. Note that some of the default arguments "
+                          "and argument names are different, so please refer "
+                          "to the documentation for `DataFrame.to_csv` when "
+                          "changing your function calls.",
+                          FutureWarning, stacklevel=2)
+
+        to_csv_kwargs = dict(path_or_buf=path, index=index, sep=sep,
+                             na_rep=na_rep, float_format=float_format,
+                             header=header, index_label=index_label,
+                             mode=mode, encoding=encoding,
+                             compression=compression,
+                             date_format=date_format,
+                             decimal=decimal)
+        to_csv_kwargs.update(**kwargs)
+
+        # Result is only a string if no path provided, otherwise None.
+        result = df.to_csv(**to_csv_kwargs)
+
         if path is None:
```

Comment:
If this fails, `None` is implicitly returned, but in that case, the value of result is also `None`. So there is no reason to test the value of `path`, it suffices to return `result` in any case.

Reply:
Sorry, I must be missing something obvious, but can you provide a valid example of calling this method with ``path=None``?

Reply:
I should think `df.to_csv()` would be a valid call, and would print the contents to screen. Maybe I am wrong.

Reply:
Yes, that is indeed a perfectly valid way to use `path=None`.